state/rds-aurora.yaml

```diff
@@ -198,7 +211,25 @@ Conditions:
   HasDBSnapshotIdentifier: !Not [!Equals [!Ref DBSnapshotIdentifier, '']]
   HasKmsKeyAndNotDBSnapshotIdentifier: !And [!Condition HasKmsKey, !Not [!Condition HasDBSnapshotIdentifier]]
   HasEngineMySQL: !Or [!Equals [!FindInMap [EngineMap, !Ref Engine, Engine], 'aurora'], !Equals [!FindInMap [EngineMap, !Ref Engine, Engine], 'aurora-mysql']]
+  HasSecretsManager: !Equals [!Ref UseSecretsManager, true]
 Resources:
+  Secret:
+    Condition: HasSecretsManager
+    Type: 'AWS::SecretsManager::Secret'
+    Properties:
+      Name: !Ref SecretName
+      GenerateSecretString:
+        SecretStringTemplate: !Sub '{"username": "${DBMasterUsername}"}'
```

Comment:
do we need the username to be part of the secret? it's never used.
like this: https://github.com/cfn-modules/secret/blob/master/module.yml#L42

Reply:
In combination with the Target Attachment the resulting secret string will look like this:
{
  "password": "superrandompassword",
  "dbname": "dbname",
  "engine": "postgres",
  "port": 5432,
  "host": "abcdefghi.abcdefghi.eu-central-1.rds.amazonaws.com",
  "username": "master"
}

I like that because my app can create the whole DB connection string from one resource.
Also I think if you want to use a rotation function it would be handy if all this information was available in the secret directly